src/hotspot/share/jvmci/jvmciRuntime.cpp

```diff
@@ -802,10 +798,10 @@ void JVMCINMethodData::set_nmethod_mirror(nmethod* nm, oop new_mirror) {
 }
 
 void JVMCINMethodData::invalidate_nmethod_mirror(nmethod* nm) {
-  oop nmethod_mirror = get_nmethod_mirror(nm, /* phantom_ref */ false);
-  if (nmethod_mirror == nullptr) {
+  if (_nmethod_mirror_index == -1) {
```

Comment:
This part is actually wrong as that's the first part of `get_nmethod_mirror` and we must always check that `get_nmethod_mirror` doesn't return nullptr.  I'd assumed that the mirror was always non-null if `_nmethod_mirror_index != -1` but that's not true.  The slot is reserved for all non-default nmethods and must stay around so that `translate` can work.